e2e/support/helpers/e2e-notebook-helpers.ts

```diff
@@ -223,3 +224,274 @@ export function selectFilterOperator(operatorName: string) {
   cy.findByLabelText("Filter operator").click();
   cy.findByRole("menu").findByText(operatorName).click();
 }
+
+type JoinType = "left-join" | "right-join" | "inner-join" | "full-join";
+
+type Stage = {
+  joins?: {
+    lhsTable: string;
+    rhsTable: string;
+    type: JoinType;
+    conditions: {
+      operator: "=" | ">" | "<" | ">=" | "<=" | "!=";
+      lhsColumn: string;
+      rhsColumn: string;
+    }[];
+  }[];
+  expressions?: string[];
+  filters?: string[];
+  aggregations?: string[];
+  breakouts?: string[];
+  sort?: {
+    column: string;
+    order: "asc" | "desc";
+  }[];
+  limit?: number;
+};
+
+export function verifyNotebookQuery(dataSource: string, stages: Stage[] = []) {
+  getNotebookStep("data").findByText(dataSource).should("be.visible");
```

Comment:
It was tempting to add even more assertions in this function (e.g. data source icon) but I think it's a good start already.